File: internship_backend_nov24/jury_solutions/cpp/c.cpp
#include <bits/stdc++.h>

int main() {
    int size, border, changes;
    std::cin >> size >> border >> changes;

    std::vector<int> elements;
    
    for (int i = 0; i < size; i++) {
        int x;
        std::cin >> x;
        elements.push_back(x);
    }
    
    int left = 0;
    int right = 0;
    int currentChanges = 0;
    int maxLength = 0;
    while (right < size) {
        if (elements[right] < border) {
            right++;
            continue;
        }

        if (currentChanges < changes) {
            currentChanges++;
            right++;
            continue;
        }

        maxLength = std::max(maxLength, right - left);

        while (elements[left] < border) 
            left++;
    
    
        left++;
        right++;
    }

    maxLength = std::max(maxLength, right - left);

    std::cout << maxLength;
}

File: internship_backend_nov24/jury_solutions/java/C.java
import java.util.Scanner;
import java.util.stream.Stream;

public class C {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int n = scanner.nextInt();
        int k = scanner.nextInt();
        int q = scanner.nextInt();

        int[] heights = Stream.generate(scanner::nextInt)
                .limit(n)
                .mapToInt(Integer::intValue)
                .toArray();

        int left = 0, changes = 0, maxLength = 0;

        for (int right = 0; right < n; right++) {
            if (heights[right] >= k) {
                changes++;
            }

            while (changes > q) {
                if (heights[left] >= k) {
                    changes--;
                }
                left++;
            }

            maxLength = Math.max(maxLength, right - left + 1);
        }

        System.out.println(maxLength);
        scanner.close();
    }
}
